### PR TITLE
also allow <adapter>.<instance-nr> for install/add cli command

### PR DESCRIPTION
**Pull request by foxriver76**
- see https://forum.iobroker.net/topic/34075/probleme-beim-javascript-update-auf-4-5-1-auf-4-6-17/31

### CODE REVIEW

File: lib/setup.js [hunk 1]
```diff
@@ -437,6 +437,17 @@ function processCommand(command, args, params, callback) {
...
+            if (!cli.tools.validateAdapterOrInstanceIdentifier(name)) {
+                showHelp();
+                return void callback(2);
+            }
+
+            // split the adapter into adapter + instance if necessary
+            if (name.includes('.')) {
+                ([name, instance] = name.split('.', 2));
+                params.instance = instance;
+            }
```
Comment (AlCalzone) [inline lib/setup.js:449]: Shouldn't we first check if instance is numeric?
This way we could write `iobroker add admin.ffff` and `instance` would be set to `ffff`, which would probably break other stuff.

File: lib/setup.js [hunk 2]
```diff
@@ -1516,7 +1527,7 @@ function processCommand(command, args, params, callback) {
             tools.getRepositoryFile(null, null, (_err, sources, _sourcesHash) => {
                 if (sources) {
                     for (const s in sources) {
-                        if (sources.hasOwnProperty(s)) {
+                        if (Object.prototype.hasOwnProperty.call(sources, s)) {
```
Comment (AlCalzone) [inline lib/setup.js:1530]: For future updates we could start changing these constructs to one of
* `for (const [s, source] of Object.entries(sources))`
* `for (const s of Object.keys(sources))`
* `for (const source of Object.values(sources))`

File: lib/setup.js (hunk 1 above)
Comment (foxriver76) [reply lib/setup.js:449]: No, see https://github.com/ioBroker/ioBroker.js-controller/pull/923/files#diff-f5d8c0f33aa88aa6e119a0cdd5f1c40eR440

File: lib/setup.js (hunk 2 above)
Comment (foxriver76) [reply lib/setup.js:1530]: :+1:

**Pull request merged**